thorlcr/activities/enth/thenthslave.cpp

```diff
@@ -103,11 +104,23 @@ class BaseEnthActivity : public CSlaveActivity, implements ILookAheadStopNotify
     {
         ActivityTimer s(totalCycles, timeActivities);
         PARENT::start();
+        lookAhead = nullptr;
         IHThorEnthArg *helper = static_cast <IHThorEnthArg *> (queryHelper());
         counter = 0;
         denominator = validRC(helper->getProportionDenominator());
         numerator = validRC(helper->getProportionNumerator());
     }
+    virtual void stop() override
+    {
+        PARENT::stop();
+
+        // restore original inputStream if lookAhead was installed, to avoid base start spuriously starting previously installed lookahead
+        if (lookAhead)
```

Comment:
trivial: This could be unconditional.  I'm not sure if it is clearer conditional or unconditional.